### ISSUE TITLE
Printing

### ISSUE THREAD

**Issue by ctgraham** [RockefellerArchiveCenter/dimes#733]
## Is your feature request related to a problem? Please describe.
Archives staff report wanting to be able to facilitate patrons printing a finding aid.

## Describe the solution you'd like
Provide a printing link which triggers request broker to [pull a PDF of the resource via API](https://archivesspace.github.io/archivesspace/api/#get-a-pdf-representation-of-a-resource).

## Describe alternatives you've considered
The browser's built in print function could be enhanced, but would be a lot of extra work, and would not be as rich as the ASpace pdf function.

## Additional context

tagging: @rzhang152

**Comment by helrond** [comment]
Just to clarify - is the assumption here that the PDF will already exist at some public URL, or will DIMES need to generate the PDF on demand?

**Comment by helrond** [comment]
We should do some testing to check the performance of this ArchivesSpace endpoint, particularly for large resources. 

In terms of integration, DIMES currently does not talk directly to ArchivesSpace (and it can't do that over JS without exposing credentials) so we'd have to go through the Request Broker. Based on my comment above, it may be wise to consider pre-generating PDFs and having them available at a URL somewhere.

**Comment by ctgraham** [comment]
Perhaps if we pre-generate the PDFs, this could be essentially another IIIF delivery?

**Comment by helrond** [comment]
I like that idea!

Side note - we will probably not use this because in many cases our collections are huge and are comprised of many AS resource records, so there's not an easy way for us to generate a PDF.

**Comment by rzhang152** [comment]
With regard to the performance concern, might be possible to add logic option selection (max# of collections or specific types ...) for limitation when generating a pdf, especially from a large resource?

**Comment by helrond** [comment]
I don't believe ArchivesSpace knows how long a PDF will take to generate via that endpoint, so we'd have to figure out some way to measure it. My understanding is that it's not just based on the number of archival objects in the resource tree, but also the structure of the tree. More complex/deeply nested trees take longer to export.

For my money, the simplest and most performant option is to pre-generate PDFs and have them waiting to be downloaded.

**Comment by rzhang152** [comment]
Agree. Pre-generating here is a better option thinking about the factor of the structure complexity.
